src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1178,6 +1183,8 @@ private async Task<WebResponse> SendRequest(bool async)
                     request.Headers.ConnectionClose = true;
                 }
 
+                request.Headers.ExpectContinue = _servicePoint?.Expect100Continue ?? true;
```

Comment:
```suggestion
                if (_servicePoint?.Expect100Continue == true)
                {
                    request.Headers.ExpectContinue = true;
                }
```

This would flip the default for Expect100Continue back to `true`.
AFAIR it used to be true on Framework, but not on Core.

Given WebRequest is considered obsolete, I think we should avoid making breaking changes here unless there is a good reason for it.

Reply:
It was false on implementation, but it wasn't according to the documentation. Since we're using this property now, it will be set to `true` all the time, but I don't have any strong argument against what you suggested, but if we keep it as `false`. We need to update the [doc](https://learn.microsoft.com/en-us/dotnet/api/system.net.servicepoint.expect100continue?view=net-8.0) as well.

Reply:
My preference would be with sticking to the current default `false` as it matches that of `SocketsHttpHandler`. Updating the docs also seems like goodness since it sounds like they were misleading for a while already